packages/shared/src/features/common/components/MobileSocialRegistration.tsx

```diff
@@ -0,0 +1,97 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import type { ButtonProps } from '../../../components/buttons/Button';
+import { Button, ButtonVariant } from '../../../components/buttons/Button';
+import { isIOSNative } from '../../../lib/func';
+import { providerMap, SocialProvider } from '../../../components/auth/common';
+import { useLogContext } from '../../../contexts/LogContext';
+import { AuthEventNames } from '../../../lib/auth';
+import {
+  Typography,
+  TypographyColor,
+  TypographyTag,
+  TypographyType,
+} from '../../../components/typography/Typography';
+import { cookiePolicy, termsOfService } from '../../../lib/constants';
+import { anchorDefaultRel } from '../../../lib/strings';
+
+interface MobileSocialRegistrationProps {
+  onClick: (provider: SocialProvider) => void;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+}
+
+export function MobileSocialRegistration({
+  onClick,
+  isLoading,
+  isDisabled,
+}: MobileSocialRegistrationProps): ReactElement {
+  const { logEvent } = useLogContext();
+  const isIosNative = isIOSNative();
+  const firstProvider = isIosNative
+    ? SocialProvider.Apple
+    : SocialProvider.Google;
```

Comment:
These are the wrong rules for the provider list. The rules are:
* If not an in-app browser (we have a function to check that), we show google
* If an in-app browser on ios device (not ios native), we show apple
* Otherwise we show facebook